docs/writing-docs/tutorials/basics.md

```diff
@@ -0,0 +1,216 @@
+```
+
+>--or--
+
+```markdown
+## Flash all the LEDs on and off twice
+```
+
+The editor automatically parses the markdown and populates the user interface from each step section.
+
+### ~ hint
+
+#### Simple, short descriptions
+
+During an interaction, the step description (all text before the first code block or image) is shown in the caption. If the paragraph length goes beyond the display length of caption, a "More" button appears in order to view the rest of the paragraph. It's best to keep the paragraph short enough to so all of it appears in the caption without requiring the user to click to see it all. If your instructions need more text, you can just create an additional step to split up the activity.
+
+### ~
+
+### Hints
+
+Hints provide additional information and code suggestions to help the user complete the step.
+In each step, any text before the first code snippet or image is automatically displayed to the user in the tutorial caption. The remaining text, block examples, etc. are displayed in the ``hint`` dialog when the user clicks the caption or hint button.
+
+### Images
+
+Using images in steps is a simple and powerful way to reinforce concepts and convey ideas. Images can be included as part of step descriptions. You specify an image using the standard markdown format.
+
+```markdown
+![Agent building a tower](/static/tutorials/agent-tower.png)
+```
+
+A step with and image might have text like this:
+
+```markdown
+## Introduction @unplugged
+
+Have the agent build a tower! Make a command to tell it how many levels to build.
+
+![Agent building a tower](/static/tutorials/agent-tower.png)
+```
+
+Images appear in steps that have either the [fullscreen](#fullscreen) or [unplugged](#unplugged) modifiers present, or when a step hint is viewed.
+
+Images should be hosted under the ``./docs/static/`` folder path of the editor project's repository files. The relative URL in markdown begins with just ``/static/`` though.
+
+Image formats typically used are PNG, JPEG, and GIF.
+
+### ~ alert
+
+#### Image size considerations
+
+It's important to consider that image size can affect the load time and therefore
+impact the experience of the user during a tutorial. Also, images that are too large can consume the viewing area where the tutorial is displayed. Try to economize tutorial images both in file size and dimensionally.
+Recommendations for images are:
+
+* **File size**: 1 MB or less
+* **Dimensions**: 800 pixels wide or less
+
+### ~
+
+## Step modifiers
+
+To add a special behavior to a step, use a step modifier:
+
+### fullscreen
+
+If you want to include a dramatic introduction or make certain that a special message is seen, you can use the ``@fullscreen`` tag. The section is displayed in an overlay window on top of the tutorial screen and isn't shown in the caption as part of the tutorial flow. You include it in your tutorial like this:
+
+```markdown
+# Flash-a-rama
+
+## It's time to code! @fullscreen
```

Comment:
so "fullscreen" auto-pops the hint and "unplugged" displays the full screen modal (and auto-advances to the next step).... maybe we should rename these